com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -165,40 +215,54 @@ var EngineLoader = {
     loadWasmAsync: function(exeName) {
         Module.instantiateWasm = function(imports, successCallback) {
             if (EngineLoader.stream_wasm && (typeof WebAssembly.instantiateStreaming === "function")) {
-                EngineLoader.streamAndInstantiateWasmAsync(exeName + ".wasm", 10, 50, imports, successCallback);
+                EngineLoader.streamAndInstantiateWasmAsync(exeName + ".wasm", imports, successCallback);
             }
             else {
-                EngineLoader.loadAndInstantiateWasmAsync(exeName + ".wasm", 10, 50, imports, successCallback);
+                EngineLoader.loadAndInstantiateWasmAsync(exeName + ".wasm", imports, successCallback);
             }
             return {}; // Compiling asynchronously, no exports.
         };
-        EngineLoader.loadAndRunScriptAsync(exeName + '_wasm.js', EngineLoader.wasmjs_size, 0, 10);
+        EngineLoader.loadAndRunScriptAsync(exeName + '_wasm.js');
     },
 
     loadAsmJsAsync: function(exeName) {
-        EngineLoader.loadAndRunScriptAsync(exeName + '_asmjs.js', EngineLoader.asmjs_size, 0, 50);
+        EngineLoader.loadAndRunScriptAsync(exeName + '_asmjs.js');
     },
 
     // load and start engine script (asm.js or wasm.js)
-    loadAndRunScriptAsync: function(src, estimatedSize, fromProgress, toProgress) {
-        FileLoader.load(src, "text", estimatedSize,
-            function(loaded, total) { Progress.calculateProgress(fromProgress, toProgress, loaded, total); },
+    loadAndRunScriptAsync: function(src) {
+        FileLoader.load(src, "text",
+            function(delta) {
+                ProgressUpdater.updateCurrent(delta);
+            },
             function(error) { throw error; },
             function(response) {
                 var tag = document.createElement("script");
                 tag.text = response;
                 document.body.appendChild(tag);
+            },
+            function(loadedDelta, currentAttempt){
+                ProgressUpdater.updateCurrent(-loadedDelta);
             });
     },
 
-    // load engine (asm.js or wasm.js + wasm)
+    // left as entrypoint for backward capability
+    // start loading archive_files.json
+    // after receiving it - start loading engine and data concurrently
     load: function(appCanvasId, exeName) {
```

Comment:
I left `EngineLoader.load` function as entrypoint because it may produce breaking change for developer.